pyphi/actual.py

```diff
@@ -410,14 +410,23 @@ def find_causal_link(self, direction, mechanism, purviews=False,
         if not purviews:
             max_ria = _null_ac_ria(self.mechanism_state(direction),
                                    direction, mechanism, None)
+            return CausalLink(max_ria)
         else:
-            # This max should be most positive
-            max_ria = max(self.find_mip(direction, mechanism, purview,
-                                        allow_neg)
-                          for purview in purviews)
-
-        # Construct the corresponding CausalLink
-        return CausalLink(max_ria)
+            # Finds rias with maximum alpha
+            all_ria = [self.find_mip(direction, mechanism, purview,allow_neg)
+                          for purview in purviews]
+            max_alpha = max(all_ria).alpha
+            max_rias = [ria for ria in all_ria if ria.alpha==max_alpha]
+
+            # Selected rias whose purview is not a superset of any other
+            purviews = [ria.purview for ria in max_rias]
+            extended_purview = []
+            for purview in purviews:
+                is_not_superset = not np.any([set(purview).issuperset(set(purview2)) and
+                    set(purview) > set(purview2) for purview2 in purviews])
+                if is_not_superset:
+                    extended_purview.append(purview)
+            return CausalLink(max(max_rias),tuple(extended_purview))
```

Comment:
The code style needs to be cleaned up here. In general, PyPhi tried to adhere to the [Google Python style guide](https://google.github.io/styleguide/pyguide.html)—but more importantly, you should always conform to the style used throughout the rest of the code.

Reply:
Installing `pylint` and running it on your code will detect problems automatically.

Reply:
Also, I would use `filter()` here, rather than using the imperative style (*i.e.*, `for` and `append()`).